src/baseclasses/chemical_energy/nome_cp_analysis.py

```diff
@@ -0,0 +1,145 @@
+
+from nomad.metainfo import Quantity, Reference, Section, SubSection, Datetime
+from nomad.datamodel.metainfo.basesections import Analysis, SectionReference, AnalysisResult, CompositeSystemReference
+from nomad.datamodel.results import Results, Material
+
+from baseclasses.chemical_energy import Chronopotentiometry
+from ..helper.utilities import get_reference
+
+
+class CPOERReference(SectionReference):
+    reference = Quantity(
+        type=Reference(Chronopotentiometry.m_def),
+        a_eln=dict(component='ReferenceEditQuantity', label='OER Chronopotentiometry Measurement'))
+
+
+class CPOERAnalysisResult(AnalysisResult):
+    voltage_avg_first5 = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    voltage_avg_last5 = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    voltage_difference = Quantity(
+        type=np.dtype(np.float64),
+        unit=('V'))
+    j = Quantity(
+        type=np.dtype(np.float64),
+        description='current density',
+        unit=('A/cm^2'))
+    current_density_string = Quantity(
+        type=str,
+        description='string representation of j (needed for terms in explore view)',)
+    experiment_duration = Quantity(
+        type=np.dtype(np.float64),
+        unit=('hr'))
+    reaction_type = Quantity(
+        type=str,
+        description='At the moment only OER CP is supported. In the future maybe also NRR CP.')
+    samples = SubSection(
+        section_def=CompositeSystemReference,
+        repeats=True,)
+
+    def normalize(self, archive, logger):
+        self.current_density_string = format(self.j, '~')
+        super(CPOERAnalysisResult, self).normalize(archive, logger)
+
+
+class CPAnalysis(Analysis):
+    m_def = Section(label_quantity='name')
+
+    inputs = Analysis.inputs.m_copy()
+    inputs.section_def = CPOERReference
+
+    outputs = Analysis.outputs.m_copy()
+    outputs.section_def = CPOERAnalysisResult
+
+    def normalize(self, archive, logger):
+        refs = find_all_cp_in_upload(archive, archive.metadata.upload_id)
+        self.inputs = [CPOERReference(reference=ref) for ref in refs]
+
+        if self.inputs is not None and len(self.inputs) >= 2:
+            first_oer_run = self.inputs[0].reference
+            last_oer_run = self.inputs[-1].reference
+
+            for cp_entry in self.inputs:
+                cp_entry.name = cp_entry.reference.data_file
+                if cp_entry.reference.datetime < first_oer_run.datetime:
+                    first_oer_run = cp_entry.reference
+                if cp_entry.reference.datetime > last_oer_run.datetime:
+                    last_oer_run = cp_entry.reference
+
+            voltage_avg_first5 = np.mean(np.array(first_oer_run.voltage[:5]))
+            voltage_avg_last5 = np.mean(np.array(last_oer_run.voltage[-5:]))
+            voltage_difference = voltage_avg_first5 - voltage_avg_last5
+
+            current_ma = first_oer_run.properties.step_1_current * 1000  # convert A to mA
```

Comment:
please just pints `.to(mA)` https://pint.readthedocs.io/en/stable/ to convert units

check that current is not `None`?

Reply:
Thinking about it mA is not even necessary here. I added a check for current and sample area to be not None.